openlibrary/plugins/upstream/addbook.py

```diff
@@ -193,11 +193,17 @@ def GET(self):
         if not self.has_permission():
             return safe_seeother(f"/account/login?redirect={self.path}")
 
-        i = web.input(work=None)
+        i = web.input(work=None, author=None)
         work = i.work and web.ctx.site.get(i.work)
+        author = i.author and web.ctx.site.get(i.author)
+
+        # pre-filling existing author(s) if adding new edition from existing work page
         authors = (work and work.authors) or []
         if work and authors:
             authors = [a.author for a in authors]
+        # pre-filling existing author if adding new work from author page
+        if author and author not in authors:
```

Comment:
We could potentially leverage some of the work from https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/__init__.py#L801-L802 where we normalize/dedupe authors on import... But I think this PR is good for now!